packages/vendure-plugin-invoices/src/api/invoice.controller.ts

```diff
@@ -11,17 +11,10 @@ import {
   BadRequestException,
 } from '@nestjs/common';
 import { Request, Response } from 'express';
-import {
-  Allow,
-  Ctx,
-  Logger,
-  Permission,
-  RequestContext,
-  UnauthorizedError,
-} from '@vendure/core';
+import { Allow, Ctx, Logger, RequestContext } from '@vendure/core';
 import { loggerCtx } from '../constants';
 import { ReadStream } from 'fs';
-import { invoicePermission } from '../index';
+import { invoicePermission } from './invoice.resolver';
```

Comment:
The `cannot read Permission of undefined` is happening when the permission is exported via a barrel file (index.ts), so  place them directly in resolvers or controllers where needed. Not sure why this wasn't the case in V1